src/wallet/rpcwallet.cpp

```diff
@@ -1687,6 +1687,39 @@ UniValue gettransaction(const UniValue& params, bool fHelp)
     return entry;
 }
 
+UniValue abandontransaction(const UniValue& params, bool fHelp)
+{
+    if (fHelp || params.size() != 1)
+        throw std::runtime_error(
+            "abandontransaction \"txid\"\n"
+            "\nMark in-wallet transaction <txid> as abandoned\n"
+            "This will mark this transaction and all its in-wallet descendants as abandoned which will allow\n"
+            "for their inputs to be respent.  It can be used to replace \"stuck\" or evicted transactions.\n"
+            "It only works on transactions which are not included in a block and are not currently in the mempool.\n"
+            "It has no effect on transactions which are already conflicted or abandoned.\n"
+            "\nArguments:\n"
+            "1. \"txid\"    (string, required) The transaction id\n"
+            "\nResult:\n"
+            "\nExamples:\n"
+            + HelpExampleCli("abandontransaction", "\"1075db55d416d3ca199f55b6084e2115b9345e16c5cf302fc80e9d5fbf5d48d\"")
+            + HelpExampleRpc("abandontransaction", "\"1075db55d416d3ca199f55b6084e2115b9345e16c5cf302fc80e9d5fbf5d48d\"")
+        );
+
+    EnsureWalletIsUnlocked();
+
+    LOCK2(cs_main, pwalletMain->cs_wallet);
```

Comment:
A little scope creep after looking at rpcwallet.cpp as a whole; as looks like most of these commands will likely crash the wallet if issued from -cli too early, since pwalletMain isn't checked before dereferencing.

Reply:
Nice catch. I'll fix that.

Reply:
looks like there's some infrastructure that I didn't know about and need to dig up:

```
spectre@Fortune:~/git-repos/PIVX/src/wallet$ ../pivx-cli getaccountaddress ""
error: {"code":-32601,"message":"Method not found (disabled)"}
```
that prevents those rpcwallet commands from crashing